src/hooks/algolia/context.tsx

```diff
@@ -0,0 +1,67 @@
+import React, {
+  createContext,
+  PropsWithChildren,
+  useContext,
+  useMemo,
+} from "react";
+import algoliasearch, { SearchClient, SearchIndex } from "algoliasearch/lite";
+
+import { ALGOLIA_API_SEARCH_KEY } from "secrets";
+
+import { ALGOLIA_APP_ID } from "settings";
+
+import { AlgoliaSearchIndex, algoliaSearchIndexes } from "types/algolia";
+
+export interface AlgoliaSearchContextState {
+  client: SearchClient;
+  indices: Record<AlgoliaSearchIndex, SearchIndex>;
+}
+
+const initState = (appId: string, key: string): AlgoliaSearchContextState => {
+  const client = algoliasearch(appId, key);
+  const indices = Object.assign(
+    {},
+    ...algoliaSearchIndexes.map((indexName: AlgoliaSearchIndex) => {
+      const index = client.initIndex(indexName);
+      return {
+        [indexName]: index,
+      };
+    })
+  );
```

Comment:
There is a lodash method that turns arrays into objects. Please, use that

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/2285/commits/591237fbb78709a84f54371437645023748f2b54